### PR TITLE
chore: roll to 1.23

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: src/Playwright.Tests/BrowserContextLocaleTests.cs [hunk 1]
```diff
@@ -42,18 +42,18 @@ public async Task ShouldAffectAcceptLanguageHeader()
                 requestTask,
                 page.GotoAsync(Server.EmptyPage));
 
-            Assert.That(acceptLanguage, Does.StartWith("fr-CH"));
+            Assert.That(acceptLanguage, Does.StartWith("fr-FR"));
```
Comment (mxschmitt) [inline src/Playwright.Tests/BrowserContextLocaleTests.cs:45]: s/fr-CH/fr-FR: are from here: https://github.com/microsoft/playwright/issues/14709

File: src/Playwright.Tests/BrowserContextServiceWorkerPolicyTests.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * SOFTWARE.
+ */
+
+using System.Threading.Tasks;
+using Microsoft.Playwright.NUnit;
+using NUnit.Framework;
+
+namespace Microsoft.Playwright.Tests
+{
+    public class BrowserContextServiceWorkerPolicyTests : BrowserTestEx
```
Comment (mxschmitt) [inline src/Playwright.Tests/BrowserContextServiceWorkerPolicyTests.cs:31]: https://github.com/microsoft/playwright/commit/85d03714d9d61dde41a05f7c54acae788f4f84c0

File: src/Playwright.Tests/BrowserTests.cs [hunk 3]
```diff
@@ -45,6 +45,16 @@ public async Task ShouldCreateNewPage()
             Assert.That(browser.Contexts, Has.Length.EqualTo(1));
 
             await page2.CloseAsync();
+            await browser.CloseAsync();
+        }
+
+        [PlaywrightTest("browser.spec.ts", "should return browserType")]
+        public async Task ShouldReturnBrowserType()
```
Comment (mxschmitt) [inline src/Playwright.Tests/BrowserTests.cs:52]: https://github.com/microsoft/playwright/commit/4c2fc6b6ebf865346b2153611d9e04b3f2abf3ac

File: src/Playwright.Tests/DefaultBrowsercontext2Tests.cs [hunk 4]
```diff
@@ -229,40 +229,6 @@ await using (var browserContext2 = await BrowserType.LaunchPersistentContextAsyn
             }
         }
 
-        [PlaywrightTest("defaultbrowsercontext-2.spec.ts", "should restore cookies from userDataDir")]
```
Comment (mxschmitt) [inline src/Playwright.Tests/DefaultBrowsercontext2Tests.cs:232]: https://github.com/microsoft/playwright/pull/14724#event-6774053878

File: src/Playwright.Tests/PageNetworkResponseTests.cs [hunk 5]
```diff
@@ -255,5 +254,22 @@ public async Task ShouldBehaveTheSameWayForHeadersAndAllHeaders()
             Assert.AreEqual(response.Headers, allHeaders);
 #pragma warning restore 0612
         }
+
+        [PlaywrightTest("page-network-response.spec.ts", "should report if request was fromServiceWorker")]
+        public async Task ShouldReportIfRequestWasFromServiceWorker()
```
Comment (mxschmitt) [inline src/Playwright.Tests/PageNetworkResponseTests.cs:259]: https://github.com/microsoft/playwright/commit/fccee89b4129d421b8a43764b2ffce02ce7c4747

File: src/Playwright.Tests/PageRequestInterceptTests.cs [hunk 6]
```diff
@@ -70,6 +70,9 @@ await route.FulfillAsync(new()
                     Response = response,
                     Status = 201,
                     Body = "",
+                    Headers = new Dictionary<string, string> {
```
Comment (mxschmitt) [inline src/Playwright.Tests/PageRequestInterceptTests.cs:73]: this fixes tests, was also made upstream the same way. 

Class got renamed to align it with file name which is common in c# and we do it everywhere else.

File: src/Playwright/Core/Browser.cs [hunk 7]
```diff
@@ -118,7 +119,6 @@ public async Task<IBrowserContext> NewContextAsync(BrowserNewContextOptions opti
                forcedColors: options.ForcedColors).ConfigureAwait(false)).Object;
 
             context.Options = options;
-            ((Tracing)context.Tracing).LocalUtils = LocalUtils;
```
Comment (mxschmitt) [inline src/Playwright/Core/Browser.cs:121]: https://github.com/microsoft/playwright/commit/176ab7e48bb73180db3740f50e167d258546206b

File: src/Playwright/Transport/Channels/PageChannel.cs [hunk 8]
```diff
@@ -83,12 +79,6 @@ internal override void OnMessage(string method, JsonElement? serverParams)
                 case "crash":
                     Crashed?.Invoke(this, EventArgs.Empty);
                     break;
-                case "domcontentloaded":
```
Comment (mxschmitt) [inline src/Playwright/Transport/Channels/PageChannel.cs:86]: https://github.com/microsoft/playwright/commit/cdb862767f408fa6d364179d449f562565b4a793

File: src/Playwright.Tests/BrowserContextServiceWorkerPolicyTests.cs [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+            await page.GotoAsync(Server.Prefix + "/serviceworkers/empty/sw.html");
+            var registrationResult = await page.EvaluateAsync<object>("() => window['registrationPromise']");
+            Assert.IsNotNull(registrationResult);
+            await context.CloseAsync();
+        }
+
+        [PlaywrightTest("browsercontext-service-worker-policy.spec.ts", "blocks service worker registration")]
+        public async Task ShouldBlockServiceWorkerRegistration()
+        {
+            var context = await Browser.NewContextAsync(new()
```
Comment (aslushnikov) [inline src/Playwright.Tests/BrowserContextServiceWorkerPolicyTests.cs:47]: don't forget to close the context!

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**